lib/DxcSupport/HLSLOptions.cpp

```diff
@@ -1071,6 +1071,21 @@ int ReadDxcOpts(const OptTable *optionTable, unsigned flagsToInclude,
   opts.SpirvOptions.entrypointName =
       Args.getLastArgValue(OPT_fspv_entrypoint_name_EQ);
 
+  // Options not currently implemented in the SPIR-V backend.
+  // The options checked here are non-exhaustive. A thorough audit of available
+  // options and their current compatibility is needed to generate a complete
+  // list.
+  if (Args.hasFlag(OPT_spirv, OPT_INVALID, false)) {
```

Comment:
This function is very long. Could we move these checks to a new helper function?

Reply:
Done